src/ORM/FieldType/DBDecimal.php

```diff
@@ -126,7 +126,7 @@ public function prepValueForDB($value)
             return 0;
         }
 
-        if (ctype_digit((string) $value)) {
+        if (abs((float) $value - (int) $value) < PHP_FLOAT_EPSILON) {
```

Comment:
Can you please explain what's happening here? It looks a bit like magic.

Reply:
The problem was between converting to string (which rounds the number correctly if it has too many decimals) and converting to int (which cuts off the decimals) which produces wrong results for numbers with fifteen or more 9's.

With this change we don't rely on the conversion to string cutting all the decimal places out but check whether the difference between the float value and the cut-off int value is smaller than the smallest representable positive number PHP can process (positive, hence using the `abs` call as well) and only then convert it to int/round it.